packages/esm-ward-app/src/ward-workspace/admit-patient-form-workspace/admit-patient-form.workspace.tsx

```diff
@@ -92,8 +90,14 @@ const AdmitPatientFormWorkspace: React.FC<AdmitPatientFormWorkspaceProps> = ({
             if (response.ok) {
               if (bedSelected) {
                 return assignPatientToBed(values.bedId, patient.uuid, response.data.uuid);
+              } else {
+                const bed = wardPatientGrouping.bedLayouts.find((bedLayout) =>
+                  bedLayout.patients.some((p) => p.uuid == patient.uuid),
+                );
+                if (bed) {
+                  return removePatientFromBed(bed.bedId, patient.uuid);
+                }
               }
-              return response;
```

Comment:
This may not be needed, but there is still a path where this line could be reached, I think (patient was not assigned to a bed and also is not in a bed on the current ward)?

Reply:
Good point. I'll leave it in.